sbin/shipper_utils/update_and_install_workers.py

```diff
@@ -52,6 +52,34 @@ def update_docker_images(user, host, worker, autograding_workers, autograding_co
     if host == "localhost":
         get_sysinfo.print_distribution()
         client = docker.from_env()
+        try:
+            # List all images
+            image_set = {
+                image_name
+                for image in client.images.list()
+                for image_name in image.attrs["RepoTags"]
+            }
+
+            images_to_remove = set.difference(image_set, images_to_update)
+
+            # Prevent removal of system docker containers
+            with open("/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/data/system_docker_containers.json") as json_file:
```

Comment:
We should not be referencing files in the `GIT_CHECKOUT` directory directly in source code for things like this.